src/frontend/screens/Settings/CreateTestData.tsx

```diff
@@ -0,0 +1,292 @@
+import {useMutation, useQueryClient} from '@tanstack/react-query';
+import {lengthToDegrees} from '@turf/helpers';
+import {randomPosition} from '@turf/random';
+import {LocationObject} from 'expo-location';
+import {type BBox} from 'geojson';
+import React, {forwardRef} from 'react';
+import {Controller, useForm} from 'react-hook-form';
+import {StyleSheet, TextInput, ToastAndroid, View} from 'react-native';
+import {UIActivityIndicator} from 'react-native-indicators';
+
+import {useActiveProject} from '../../contexts/ActiveProjectContext';
+import {useApi} from '../../contexts/ApiContext';
+import {OBSERVATION_KEY} from '../../hooks/server/observations';
+import {useLocation} from '../../hooks/useLocation';
+import {LIGHT_GREY, RED, WHITE} from '../../lib/styles';
+import {Button} from '../../sharedComponents/Button';
+import {LocationView} from '../../sharedComponents/Editor/LocationView';
+import {ScreenContentWithDock} from '../../sharedComponents/ScreenContentWithDock';
+import {Text} from '../../sharedComponents/Text';
+
+const DISTANCE_BUFFER_KM = 50;
+
+const BASE_NUMBER_INPUT_RULES = {
+  min: 1,
+};
+
+export function CreateTestDataScreen() {
+  const locationState = useLocation({maxDistanceInterval: 0});
+  const createFakeObservations = useCreateFakeObservationsMutation();
+
+  const {
+    control,
+    handleSubmit,
+    formState: {errors},
+  } = useForm<{count?: number; distance?: number}>({
+    mode: 'onBlur',
+    shouldFocusError: false,
+  });
+
+  return (
+    <ScreenContentWithDock
+      contentContainerStyle={styles.contentContainer}
+      dockContent={
+        <Button
+          fullWidth
+          disabled={createFakeObservations.status === 'pending'}
+          onPress={handleSubmit(data => {
+            if (data.count === undefined) return;
+            if (!locationState.location) {
+              ToastAndroid.show('Waiting for location', ToastAndroid.SHORT);
+              return;
+            }
+
+            createFakeObservations.mutate(
+              {
+                count: data.count,
+                location: locationState.location,
+                distance:
+                  data.distance === undefined
+                    ? DISTANCE_BUFFER_KM
+                    : data.distance,
+              },
+              {
+                onSuccess: () => {
+                  ToastAndroid.show('Observations created', ToastAndroid.SHORT);
```

Comment:
i think we should also just navigate back to the home screen here as well

Reply:
going to leave out for now but can easily follow up if QA asks for it